tsconfig.json

```diff
@@ -10,13 +10,10 @@
     "strict": true,
     "declaration": true,
     "declarationMap": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "useUnknownInCatchVariables": false
```

Comment:
Is this related?

Reply:
yea it's required for the typedoc docusaurus plugin to work.

if this flag isn't enabled, the plugin throws the following error when it runs:

<details>
<summary>error msg</summary>

```
Error: ../src/bluehawk/actions/copy.ts:29:62 - error TS2571: Object is of type 'unknown'.

29     const message = `Could not load stats for ${rootPath}: ${error.message}`;
                                                                ~~~~~

Error: ../src/bluehawk/actions/copy.ts:50:76 - error TS2571: Object is of type 'unknown'.

50       const message = `Failed to copy file ${filePath} to ${targetPath}: ${error.message}`;
                                                                              ~~~~~

Error: ../src/bluehawk/actions/copy.ts:111:100 - error TS2571: Object is of type 'unknown'.

111       const message = `Failed to write file ${targetPath} (based on ${parseResult.source.path}): ${error.message}`;
                                                                                                       ~~~~~

Error: ../src/bluehawk/actions/snip.ts:162:95 - error TS2571: Object is of type 'unknown'.

162       const message = `Failed to write ${targetPath} (based on ${parseResult.source.path}): ${error.message}`;
                                                                                                  ~~~~~

Error: ../src/bluehawk/bluehawk.ts:132:3 - error TS2571: Object is of type 'unknown'.

132 ${e.stack}
      ~

Error: ../src/bluehawk/bluehawk.ts:162:65 - error TS2571: Object is of type 'unknown'.

162         `falling back to plaintext parser for ${source.path}: ${error.message}`
                                                                    ~~~~~

Error: ../src/bluehawk/parser/visitor/makeCstVisitor.ts:244:37 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Error'.

244             jsonErrorToVisitorError(error, json, {
                                        ~~~~~

Error: ../src/cli/cli.ts:45:7 - error TS2571: Object is of type 'unknown'.

45       error.message = `Plugin '${path}' register() failed with error: ${error.message}`;
         ~~~~~

Error: ../src/cli/cli.ts:45:73 - error TS2571: Object is of type 'unknown'.

45       error.message = `Plugin '${path}' register() failed with error: ${error.message}`;
```

</details>

Reply:
Looks like these are errors in our (Bluehawk) code. Do you think we could update bh to cast the `unkown` errors instead (in the future perhaps)?

`unknown` is technically more correct and in general I think it's best that we stick as closely as possible to default/strict typescript.

Reply:
i don't really understand the question, could you expand a bit?

Reply:
I'm mainly interested if the errors you see with `{ useUnknownInCatchVariables: true }` are something we could fix by updating the bluehawk source code. If so, I think we should fix them (i.e. [what they do on this page](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables))

Reply:
I fixed this on our end in https://github.com/mongodb-university/Bluehawk/pull/76

Once we merge that can we try docusaurus with `useUnknownInCatchVariables` set to `true`?